### DISCUSSION TITLE
Declarative vs db = SQLAlchemy(app)

### DISCUSSION THREAD

**Discussion by cpdivers** [sqlalchemy/sqlalchemy#7932] [Usage Questions] [answered]
Hello guys,

surely one of my basic question again, but can someone explain to me what is the difference, pros and cons of this 2 pieces of code:
And what are they called ? The first one I believe is called Declarative, but that's the name of the other type ?

```
from sqlalchemy.ext.declarative import declarative_base

Base = declarative_base()
metadata = Base.metadata


class BLCategory(Base):
    __tablename__ = 'BL_categories'

    category_id = Column(INTEGER(11), primary_key=True, server_default=text("0"))
    category_name = Column(String(40, 'utf8mb4_unicode_ci'), nullable=False)
```

and

```
from flask_sqlalchemy import SQLAlchemy
db = SQLAlchemy(app)

class BLCategory(db.Model):
    __tablename__ = 'BL_categories'

    category_id = Column(INTEGER(11), primary_key=True, server_default=text("0"))
    category_name = Column(String(40, 'utf8mb4_unicode_ci'), nullable=False)
  ```

**Comment by CaselIT** [comment, marked as answer]
Hi,

Not an expert in it, but the second is flask-sqlalchemy that's an external library that's not developed by sqlalchemy, so it has no name in the docs.
It can be used to integrate sqlalchemy with flask application so it's usually used only when using flask. (But I believe you can also use sqlalchemy with flask without flask-sqlalchemy)
